### PR TITLE
Add CodeStyle option to allow users to customize the 'Invoke Delegate With Conditional Access' analyzer.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide  @Pilchie @dpoeschl 

Code Options look like this:
image

Feature looks like this now:

image

image

**Comment by CyrusNajmabadi** [comment]
This will also be used for: https://github.com/dotnet/roslyn/pull/13891

You can see some of the pieces of hte code for that here.

File: src/Features/CSharp/Portable/InvokeDelegateWithConditionalAccess/InvokeDelegateWithConditionalAccessAnalyzer.cs [hunk 1]
```diff
@@ -18,25 +19,44 @@ internal static class Constants
...
+
         public override void Initialize(AnalysisContext context)
         {
             context.RegisterSyntaxNodeAction(SyntaxNodeAction, SyntaxKind.IfStatement);
         }
 
         private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
         {
+            var options = syntaxContext.Options.GetOptionSet();
+            var styleOption = options.GetOption(CSharpCodeStyleOptions.PreferConditionalFunctionCall);
```
Comment (DustinCampbell) [inline src/Features/CSharp/Portable/InvokeDelegateWithConditionalAccess/InvokeDelegateWithConditionalAccessAnalyzer.cs]: The option name seems a little non-specific. Maybe "PreferConditionalDelegateCall"?

File: src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/DocumentBasedFixAllProvider.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            // Process all documents in parallel.
+            var updatedDocumentTasks = documentsAndDiagnosticsToFixMap.Select(
+                kvp => FixDocumentAsync(kvp.Key, kvp.Value, cancellationToken));
+
+            await Task.WhenAll(updatedDocumentTasks).ConfigureAwait(false);
+
+            var currentSolution = fixAllState.Solution;
+            foreach (var task in updatedDocumentTasks)
+            {
+                var updatedDocument = await task.ConfigureAwait(false);
```
Comment (DustinCampbell) [inline src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/DocumentBasedFixAllProvider.cs]: You awaited all of the tasks above. Can't you just grab the Result here?

File: src/Features/CSharp/Portable/InvokeDelegateWithConditionalAccess/InvokeDelegateWithConditionalAccessAnalyzer.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/InvokeDelegateWithConditionalAccess/InvokeDelegateWithConditionalAccessAnalyzer.cs]: Fixed.

**Comment by alrz** [comment]
I think `this` inside `if` shouldn't be faded out as it remains after applying the fix.

**Comment by CyrusNajmabadi** [comment]
"this" is being faded by another fix.

**Pull request merged**